app/payments/RazorpayPayment.php

```diff
@@ -0,0 +1,164 @@
+    {
+        $data = array(
+            'receipt' => $orderId,
+            'amount' => fn_rzp_adjust_amount($orderInfo['total'], $processorData['processor_params']['currency'])*100,
+            'currency' => $processorData['processor_params']['currency'],
+            'payment_capture' => 1
+        );
+
+        return $data;
+    }
+
+    public function generateHtmlForm($url, $json)
+    {
+        $html = <<<EOT
+<script src="https://checkout.razorpay.com/v1/checkout.js"></script>
+<script>
+    var data = $json;
+    data.handler = function (transaction) {
+        document.getElementById('razorpay_payment_id').value = transaction.razorpay_payment_id;
+        document.getElementById('razorpay_signature').value = transaction.razorpay_signature;
+        document.getElementById('razorpay-form').submit();
+    };
+    function razorpaySubmit(){                  
+        var rzp1 = new Razorpay(data);
+        rzp1.open();
+        rzp1.modal.options.backdropClose = false;
+    }    
+            
+    var rzp_interval = setInterval(function(){
+        if (typeof window["Razorpay"] != "undefined")
+        {
+            setTimeout(function(){ razorpaySubmit(); }, 500);
+            clearInterval(rzp_interval);
+        }
+    }, 500);
+</script>
+<form name="razorpay-form" id="razorpay-form" action=$url target="_parent" method="POST">
+    <input type="hidden" name="razorpay_payment_id" id="razorpay_payment_id" />
+    <input type="hidden" name="razorpay_signature" id="razorpay_signature"/>
+</form>
+</body>
+</html>
+EOT;
+
+        return $html;
+    }
+
+    public function processRazorpayResponse()
+    {
+        $razorpaySignature = $_POST['razorpay_signature'];
+
+        $razorpayPaymentId = $_POST['razorpay_payment_id'];
+
+        $merchantOrderId = fn_rzp_place_order($_SESSION['merchant_order_id']);
+
+        $razorpayOrderId = $_SESSION['razorpay_order_id'];
+
+        if ((empty($razorpaySignature) == false) and (empty($razorpayPaymentId) == false))
+        {
+            if (fn_check_payment_script('razorpay.php', $merchantOrderId, $processorData))
+            {
+                $keyId = $processorData['processor_params']['key_id'];
+
+                $keySecret = $processorData['processor_params']['key_secret'];
+
+                $orderInfo = fn_get_order_info($merchantOrderId);
+
+                $amount = fn_rzp_adjust_amount($orderInfo['total'],
+                    $processorData['processor_params']['currency'])*100;
+                    
+                $attributes = array (
+                    'razorpay_signature'  => $razorpaySignature,
+                    'razorpay_order_id'   => $razorpayOrderId,
+                    'razorpay_payment_id' => $razorpayPaymentId
+                );
+
+                $api = new Api($keyId, $keySecret);
+
+                try
+                {
+                    $success = $api->utility->verifyPaymentSignature($attributes);
```

Comment:
No need for a try-catch here. verifyPaymentSignature doesn't raise any exceptions.